api/envoy/service/auth/v3/attribute_context.proto

```diff
@@ -116,7 +117,28 @@ message AttributeContext {
     // The HTTP request headers. If multiple headers share the same key, they
     // must be merged according to the HTTP spec. All header keys must be
     // lower-cased, because HTTP header keys are case-insensitive.
-    map<string, string> headers = 3;
+    // Header value is encoded as string. This does not work for non-utf8 characters.
```

Comment:
IIUC from the other API comment non-utf8 characters are replaced by "!". If that's the case I suggest the following:

```suggestion
    // Header value is encoded as string. This does not work for non-utf8 characters, and therefore they will be replaced by "!".
```

Also please add that "Depending on the configuration, only this field or ``header_map`` will be populated in a request.".